crates/lair_keystore_api/src/lair_server/priv_api.rs

```diff
@@ -233,6 +233,106 @@ pub(crate) fn priv_req_new_seed<'a>(
+    async move {
+        let store = priv_get_store(inner, unlocked)?;
+
+        let entry = store.get_entry_by_tag(req.src_tag.clone()).await?;
+
+        let (src_seed, src_seed_info) = match (&*entry, req.src_deep_lock_passphrase) {
+            (LairEntryInner::Seed { seed, seed_info, .. }, None) => {
+                let seed = seed.decrypt(store.get_bidi_ctx_key()).await?;
+                (seed, seed_info)
+            }
+
+            (
+                LairEntryInner::DeepLockedSeed {
+                    seed,
+                    seed_info,
+                    salt,
+                    ..
+                },
+                Some(secret),
+            ) => {
+                let deep_key = deep_unlock_key_from_passphrase(
+                    &secret,
+                    dec_ctx_key,
+                    BufReadSized::from(salt.clone().cloned_inner()),
+                )
+                .await?;
+                let seed = seed.decrypt(deep_key).await?;
+                (seed, seed_info)
+            },
+
+            (LairEntryInner::WkaTlsCert { .. }, _) => return Err("The tag provided is for a Cert, which cannot be derived. You must specify the tag for a Seed.".into()),
+            (LairEntryInner::Seed { .. }, Some(_)) => return Err("A passphrase was provided for a seed which is not deep-locked. Make the request without a `src_passphrase`.".into()),
+            (LairEntryInner::DeepLockedSeed { .. }, None) => return Err("A `src_passphrase` is needed to unlock the source seed which is deep-locked.".into()),
+        };
+
+        let mut parent = src_seed;
+
+        for index in req.derivation_path.iter() {
+            let derived = BufWriteSized::new_mem_locked()?;
+            sodoken::kdf::derive_from_key(
+                derived.clone(),
+                *index as u64,
+                *b"IdnSecKy",
+                parent,
+            )?;
+            parent = derived.clone().into();
+        }
+
+        let dst_seed = parent;
+        let dst_dlp = req.dst_deep_lock_passphrase;
+
+        let dst_seed_info = match dst_dlp {
+            Some(secret) => {
+                // if deep locked, decrypt the deep lock passphrase
+                let deep_lock_passphrase =
+                    secret.passphrase.decrypt(dec_ctx_key.clone()).await?;
+
+                store
+                    .insert_deep_locked_seed(
+                        dst_seed,
+                        req.dst_tag.clone(),
+                        secret.ops_limit,
+                        secret.mem_limit,
+                        deep_lock_passphrase,
+                        src_seed_info.exportable,
+                    )
+                    .await?
+            }
+            // create a new seed
+            None => {
+                store
+                    .insert_seed(
+                        dst_seed,
+                        req.dst_tag.clone(),
+                        src_seed_info.exportable,
```

Comment:
... someday we'll probably want to be able to change the exportable and limits when deriving... but out of scope until we have an actual need for it : )